src/Badge/Infrastructure/Package/PackageRepository.php

```diff
@@ -48,11 +48,15 @@ public function fetchByRepository(string $repository): Package
 
         $repositoryInfo = Repository::createFromRepositoryUrl($apiPackage->getRepository());
 
-        $defaultBranch = $this->clientStrategy->getDefaultBranch($repositoryInfo);
-
         /** @var Package $class */
         $class = self::$packageClass;
 
-        return $class::createFromApi($apiPackage, ['default_branch' => $defaultBranch]);
+        $package = $class::createFromApi($apiPackage);
```

Comment:
IMO it would be simpler if we create the object only once.
We can do an if/else here and remove the `withDefaultBranch` method definition.

Reply:
Thanks for the suggestion. Makes absolute sense here